runner/app/live/streamer/process.py

```diff
@@ -117,91 +119,108 @@ def process_loop(self):
+                    self._report_error(f"Error loading pipeline with default params: {e}")
                     raise
 
-            while not self.is_done():
-                while not self.param_update_queue.empty():
-                    params = self.param_update_queue.get_nowait()
-                    try:
-                        logging.info(f"PipelineProcess: Processing parameter update from queue: {params}")
-                        if _handle_logging_params(params):
-                            logging.info(f"PipelineProcess: Updating pipeline parameters")
-                            pipeline.update_params(**params)
-                            logging.info(f"PipelineProcess: Successfully applied params to pipeline: {params}")
-                    except Exception as e:
-                        error_msg = f"Error updating params: {str(e)}"
-                        logging.error(error_msg, exc_info=True)
-                        report_error(error_msg)
-
-                try:
-                    input_frame = self.input_queue.get(timeout=0.1)
-                except queue.Empty:
-                    continue
+    async def _run_pipeline_loops(self):
+        pipeline = await self._initialize_pipeline()
+        await pipeline.warm_video()
+        input_task = asyncio.create_task(self._input_loop(pipeline))
+        output_task = asyncio.create_task(self._output_loop(pipeline))
+        param_task = asyncio.create_task(self._param_update_loop(pipeline))
 
-                try:
-                    if isinstance(input_frame, VideoFrame):
-                        input_frame.log_timestamps["pre_process_frame"] = time.time()
-                        output_image = pipeline.process_frame(input_frame.image)
-                        input_frame.log_timestamps["post_process_frame"] = time.time()
-                        output_frame = VideoOutput(input_frame.replace_image(output_image))
-                        self.output_queue.put(output_frame)
-                    elif isinstance(input_frame, AudioFrame):
-                        self.output_queue.put(AudioOutput([input_frame]))
-                        # TODO wire in a proper pipeline here
-                    else:
-                        report_error(f"Unsupported input frame type {type(input_frame)}")
-                except Exception as e:
-                    report_error(f"Error processing frame: {e}")
+        try:
+            await asyncio.gather(input_task, output_task, param_task)
         except Exception as e:
-            report_error(f"Error in process run method: {e}")
-        finally:
-            self._cleanup_pipeline(pipeline)
+            self._report_error(f"Error in pipeline loops: {e}")
+            await self._cleanup_pipeline(pipeline)
+            raise
+
+    async def _input_loop(self, pipeline):
+        while not self.is_done():
+            try:
+                input_frame = await asyncio.to_thread(self.input_queue.get, timeout=0.1)
```

Comment:
maybe we dont need the timeout if we're not doing anything on `queue.Empty`? i think i only did it with a timeout myself when we were in a non-async context and i didn't want to block the thread (or maybe it was even before i knew `asyncio.to_thread` existed)

Reply:
also side comment, i hope `asyncio` is really efficient and we'll not be creating a bunch of threads here

Reply:
Addressed in https://github.com/livepeer/ai-runner/commit/d6922ab24ea8a5e22e3d565500404273be361ad2